lib/web_ui/dev/browser_roller.dart

```diff
@@ -0,0 +1,318 @@
+    'verbose',
+    abbr: 'v',
+    defaultsTo: false,
+    help: 'Enable verbose output.',
+    negatable: false,
+  );
+
+late final String versionTag;
+late final String? desiredPlatform;
+late final bool dryRun;
+late final bool verbose;
+
+final Client _client = Client();
+
+/// Rolls browser CIPD packages to the version specified in `browser_lock.yaml`.
+///
+/// Currently only rolls Chrome.
+///
+/// These rolls are consumed by the "chrome_and_driver" LUCI recipe, here:
+/// https://cs.opensource.google/flutter/recipes/+/main:recipe_modules/flutter_deps/api.py;l=146
+Future<void> main(List<String> args) async {
+  try {
+    processArgs(_argParser.parse(args));
+    await _BrowserRoller().roll();
+    io.exitCode = 0;
+  } on FormatException catch (e) {
+    print('''
+Error! ${e.message}
+
+Available options:
+
+${_argParser.usage}
+''');
+    io.exitCode = 1;
+  } finally {
+    _client.close();
+  }
+}
+
+// Initialize globals from the parsed command-line arguments.
+void processArgs(ArgResults args) {
+  versionTag = args['version-tag'] as String;
+  desiredPlatform = args['platform'] as String?;
+  dryRun = args['dry-run'] as bool;
+  verbose = args['verbose'] as bool;
+}
+
+class _BrowserRoller {
+  _BrowserRoller();
+
+  final io.Directory _rollDir = io.Directory.systemTemp.createTempSync('browser-roll-');
+
+  final Map<String, PlatformBinding> _platformBindings = <String, PlatformBinding>{
+    'linux': LinuxPlatformBinding(),
+    'mac': MacPlatformBinding(),
+    'windows': WindowsPlatformBinding(),
+  };
+
+  final BrowserLock _lock = BrowserLock();
+
+  // Prints output when --verbose is set.
+  void vprint(String out) {
+    if (verbose) {
+      print(out);
+    }
+  }
+
+  // Roll Chromium and ChromeDriver for each of the Platforms.
+  Future<void> roll() async {
+    for (final MapEntry<String, PlatformBinding> entry in _platformBindings.entries) {
+      final String platform = entry.key;
+      if (desiredPlatform != null && platform != desiredPlatform) {
+        vprint('Skipped platform: $platform');
+        continue;
+      }
+      final PlatformBinding binding = entry.value;
+      await _rollChromium(platform, binding);
+      await _rollChromeDriver(platform, binding);
+    }
+    if (dryRun) {
+      print('\nDry Run! Non-published roll artifacts kept here: ${_rollDir.path}\n');
+    } else {
+      // Clean-up
+      vprint('\nDeleting temporary directory: ${_rollDir.path}');
+      await _rollDir.delete(recursive: true);
+      print('\nDone.\n');
+    }
+  }
+
+  // Returns the contents for the CIPD config required to publish a new chromium package.
+  String _getCipdChromiumConfig({
+    required String platform,
+    required String version,
+    required String build,
+    required String root,
+  }) {
+    return '''
+package: flutter_internal/browsers/chrome/$platform-amd64
+description: Chromium $version (build $build) used for testing
+preserve_writable: true
+root: $root
+data:
+  - dir: .
```

Comment:
The directory structure could be:

```
_rollDir/
  linux/
    chrome-linux/**/*
  mac/
    chrome-mac/**/*
  windows/
    chrome-windows/**/*
  linux.cipd.yaml
  mac.cipd.yaml
  windows.cipd.yaml
```

In this case the value of `root`/`data` can simply be:

```
root: $platform
data:
  - dir: .
```

Assuming CWD is `_rollDir`:

```
cipd create --pkg-config=linux.cipd.yaml ...other options...
```

Reply:
Turns out some recipes expect chrome to exist in the root of the uncompressed package, others expect it to be inside a `chrome-mac` directory.

I've placed the download in BOTH locations, until we get the recipes to agree on the location. Then we can remove from the script and the package the unneeded copy.